tests/unit/domain/sing/shouldPlayTracks.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { createDefaultTrack, shouldPlayTracks } from "@/sing/domain";
+import { Track } from "@/store/type";
+import { TrackId } from "@/type/preload";
+
+const createTrack = ({ solo, mute }: { solo: boolean; mute: boolean }) => {
+  const track = createDefaultTrack();
+
+  if (solo) {
+    track.solo = true;
+  }
+  if (mute) {
+    track.mute = true;
+  }
+
+  return track;
+};
+const toTracksMap = (tracks: Track[]) => {
+  return tracks.reduce((acc, track) => {
+    acc.set(TrackId(crypto.randomUUID()), track);
+    return acc;
+  }, new Map<TrackId, Track>());
+};
+const findIndices = (tracks: Map<TrackId, Track>, trackIds: Set<TrackId>) =>
+  [...trackIds].map((trackId) => {
+    return Array.from(tracks.keys()).findIndex((id) => id === trackId);
+  });
+
+describe("shouldPlayTracks", () => {
+  it("ソロのトラックが存在する場合はソロのトラックのみ再生する（ミュートは無視される）", () => {
+    const tracks = toTracksMap([
+      createTrack({ solo: false, mute: false }),
+      createTrack({ solo: false, mute: false }),
+      createTrack({ solo: true, mute: false }),
+      createTrack({ solo: true, mute: false }),
+      createTrack({ solo: false, mute: true }),
+      createTrack({ solo: false, mute: true }),
+    ]);
+
+    const result = shouldPlayTracks(tracks);
+    expect(findIndices(tracks, result).sort()).toEqual([2, 3]);
+  });
+
+  it("ソロのトラックが存在しない場合はミュートされていないトラックを再生する", () => {
+    const tracks = toTracksMap([
+      createTrack({ solo: false, mute: false }),
+      createTrack({ solo: false, mute: false }),
+      createTrack({ solo: false, mute: true }),
+      createTrack({ solo: false, mute: true }),
+    ]);
+
+    const result = shouldPlayTracks(tracks);
+    expect(findIndices(tracks, result).sort()).toEqual([0, 1]);
```

Comment:
（細かいけど）
`findIndices`関数の定義をやめて、idをそのままtoEqualするとちょっと直感的になるかも。
```suggestion
    const trackIds = Array.from(tracks.keys());
    const result = shouldPlayTracks(tracks);
    expect(result).toEqual(new Set([trackIds[0], trackIds[1]]));
```

あるいはこういう感じのがシンプルかもしれない。
```ts
const tracks = new Map([
  [DummyId(0), createTrack({ solo: false, mute: false })],
  [DummyId(1), createTrack({ solo: false, mute: false })],
  [DummyId(2), createTrack({ solo: false, mute: true })],
  [DummyId(3), createTrack({ solo: false, mute: true })],
]);

const result = shouldPlayTracks(tracks);
expect(result).toEqual(new Set([DummyId(0), DummyId(1)]));
```